adds youtube-sub-extractor.el recipe

agzam opened the pull request:
### Brief summary of what the package does

Extracts video subtitles using https://github.com/yt-dlp (recommended) or `youtube-dl` and shows them in a buffer

### Direct link to the package repository

https://github.com/agzam/youtube-sub-extractor.el

### Your association with the package

I'm the author

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

`youtube-sub-extractor.el` with byte-compile using Emacs 28.1:
```
youtube-sub-extractor.el:139:31: Warning: the function `cl-first' is not known
    to be defined.
```
This one simply requires a `(require 'cl-lib)`.

`youtube-sub-extractor.el` with checkdoc 0.6.2:
```
youtube-sub-extractor.el:43: First line is not a complete sentence
youtube-sub-extractor.el:189: Argument `video-url' should appear (as VIDEO-URL) in the doc string
```
I believe the first one comes up for compatibility with M-x apropos, e.g. you could write `"Whether to prompt for the language when multiple are available."`.

`youtube-sub-extractor.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- youtube-sub-extractor.el#L224: It's safer to sharp-quote function names; use `#'`
- youtube-sub-extractor.el#L246 and 193: Use `(temporary-file-directory)` instead of /tmp in code
- youtube-sub-extractor.el#L248: No `format` required; `error` takes an f-string
- youtube-sub-extractor.el#L190: No `format` required; `message` takes an f-string
```

I also notice you're calling `shell-command-to-string` in `youtube-sub-extractor--send-request` without running `shell-quote-argument` on any of the arguments; this could be dangerous for users if they use `youtube-sub-extractor-extract-subs` interactively, for example, and end up specifying a URL with shell control characters or similar.

Also your commentary has a paragraph that just says "Description"; possible inocmplete stub?

Package and license:
- Prefer equivalent recipe: `(youtube-sub-extractor :repo "agzam/youtube-sub-extractor.el" :fetcher github)`
- Add *formal* license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to youtube-sub-extractor.el

agzam commented:
Wow. Thank you for such a thorough review. Sincerely appreciated. I went through all the items you've listed and fixed them, see the two latest commits.

riscy commented on recipes/youtube-sub-extractor:
```diff
@@ -0,0 +1 @@
+(youtube-sub-extractor :fetcher github :repo "agzam/youtube-sub-extractor.el" :files ("*.el"))
```
```suggestion
(youtube-sub-extractor :repo "agzam/youtube-sub-extractor.el" :fetcher github)
```

riscy replied:
(This recipe is equivalent and gives your repo room to grow without a followup PR.)

agzam replied:
The reason for :files filter is that I wanted to exclude tests. If you think I don't have to do that, I can remove it.

riscy replied:
Check out the [default value](https://github.com/melpa/melpa/blob/master/README.md#recipe-format) for the recipe file selector: it will exclude files that end in -test.el or -tests.el and has a lot of other conventions.

riscy reviewed (commented):
Thank you for saying so. :)  You caught everything I mentioned, but I think more issues have popped out during subsequent restructuring.  Here's the latest --

`youtube-sub-extractor.el` with checkdoc 0.6.2:
```
youtube-sub-extractor.el:35: Lisp symbol `left-margin' should appear in quotes
youtube-sub-extractor.el:80: First line is not a complete sentence
youtube-sub-extractor.el:80: First line should be capitalized
youtube-sub-extractor.el:200: Argument `vid-url' should appear (as VID-URL) in the doc string
youtube-sub-extractor.el:277: All interactive functions should have documentation
youtube-sub-extractor.el:295: All interactive functions should have documentation
```
I think the one on line 35 is a false positive and you can disregard

`youtube-sub-extractor.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- youtube-sub-extractor.el#L221: Consider `point-at-bol`
- youtube-sub-extractor.el#L222: Consider `point-at-eol`
- youtube-sub-extractor.el#L87: Top-level `define-key` can overwrite bindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
- youtube-sub-extractor.el#L91: Top-level `define-key` can overwrite bindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
```
For lines 87 and 91, you you can `defvar` the keymap before the `define-minor-mode` and the minor mode will automatically use the keymap you've defvar'd.

agzam commented:
Sorry, I made some improvements for which I wanted to add checks, but it was too late and I thought of doing it later, but didn't get a chance. I fixed all new issues you've listed. Also, I have added a GH Actions job https://github.com/agzam/youtube-sub-extractor.el/blob/main/.github/workflows/tests.yml#L39 with melpazoid.

riscy commented:
Here's the latest suggestions:

`youtube-sub-extractor.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- youtube-sub-extractor.el#L222: Consider `point-at-bol`
- youtube-sub-extractor.el#L223: Consider `point-at-eol`
- youtube-sub-extractor.el#L87: Top-level `define-key` can overwrite bindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
- youtube-sub-extractor.el#L91: Top-level `define-key` can overwrite bindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
- Prefer equivalent recipe: `(youtube-sub-extractor :repo "agzam/youtube-sub-extractor.el" :fetcher github)`
```

agzam commented:
> Here's the latest suggestions:

Ah, totally missed those. How do I make the GH Action fail for melpazoid suggestions?

riscy commented:
https://github.com/riscy/melpazoid#add-it-to-github-actions - although it won't fail on those, only on checklist errors.  It looks like you caught all of them, except:

Prefer equivalent recipe: `(youtube-sub-extractor :repo "agzam/youtube-sub-extractor.el" :fetcher github)` - this will let your repo grow to  e.g. have test files that are automatically excluded from the recipe.

agzam commented:
@riscy I've changed the recipe

riscy commented:
Thanks!
